pkg/network/render.go

```diff
@@ -385,6 +388,59 @@ func isNetworkChangeSafe(prev, next *operv1.NetworkSpec, infraRes *bootstrap.Inf
 	return nil
 }
 
+func isClusterNetworkChangeSafe(prev, next *operv1.NetworkSpec) bool {
+
+	// quick check to make sure clusterNetwork slices are of same size as we do not
+	// support adding/removing additional clusterNetwork entries
+	if len(prev.ClusterNetwork) != len(next.ClusterNetwork) {
+		log.Printf("Adding/Removing clusterNetwork entries of the same type is not supported")
+		return false
+	}
+
+	// Only support changing ClusterNetwork CIDR if it's OVNK
+	if !reflect.DeepEqual(next.DefaultNetwork.Type, operv1.NetworkTypeOVNKubernetes) {
+		log.Printf("Adding/Removing clusterNetwork entries of the same type is not supported")
+		return false
+	}
+
+	// sort prev and next just in case there was some re-ordering of the slice, since we
+	// want to know for sure we are comparing the same elements in each below
+	sort.SliceStable(prev.ClusterNetwork, func(i, j int) bool {
+		return prev.ClusterNetwork[i].CIDR < prev.ClusterNetwork[j].CIDR
+	})
+	sort.SliceStable(next.ClusterNetwork, func(i, j int) bool {
+		return next.ClusterNetwork[i].CIDR < next.ClusterNetwork[j].CIDR
+	})
+
+	// since we do not allow the clusterNetwork[] size to change, it should be safe to compare
```

Comment:
`clusterNetwork` length can change in conversion from/to dual-stack https://docs.openshift.com/container-platform/4.12/networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html